sys/net/gnrc/netif/gnrc_netif_ieee802154.c

```diff
@@ -44,9 +46,17 @@ gnrc_netif_t *gnrc_netif_ieee802154_create(char *stack, int stacksize,
                                            netdev_t *dev)
 {
     return gnrc_netif_create(stack, stacksize, priority, name, dev,
+                             _bootstrap,
                              &ieee802154_ops);
 }
 
+static void *_bootstrap(void *args)
+{
+    gnrc_netif_t *netif = args;
+    netif->hwdev = netif->dev;
```

Comment:
Why don't `netif_ethernet` or `netif_raw` require this change?

Reply:
`gnrc_netif_ieee802154.c` requires access to the ieee802154_netdev struct, for the pan_id, and sequence numbers and more. The other mac layers don't require this.

Reply:
Ah!

Reply:
If we can build a getter or something for this we can omit the whole hwdev member here altogether (solving the issue above too)

Reply:
Mh... if that is possible, I would prefer that. Then most of the changes in #9330 wouldn't be necessary also.

Reply:
@miri64 I see two ways to implement this:

1. Have a netopt operation to get the ieee802154_netdev_t struct. Implement this in the `netdev_ieee802154.c` layer and let it assume that the "next" layer is always the radio.

2. have gnrc_netif_ieee802154.c Iterate over the layers and assume that the last layer is the radio it is looking for.

I'd rather not implement this in the radio drivers themselves to prevent code duplication.

Reply:
Make a separate layer for the MAC framing/parsing. This is the only reason these fields are in the netdev_ieee802154_t struct. Implementing a radio MAC layer as a netdev_layer module is not straightforward with the current architecture. To get the correct outgoing frames with the configured pan id etc it is necessary to mirror the network device's netdev_ieee802154_t fields, because the gnrc_netif send function reads it from the topmost layer. From my perspective it would be better if everything related to frame construction was forked off into its own layer, so that the netdev device drivers for transceivers don't have to carry RAM copies of the device address, PAN id, sequence number etc. It would simplify the get/set implementations on those drivers, and the drivers would just read/write the relevant hardware registers instead of having to also keep this struct up to date.

Reply:
By "make a separate layer" I mean move the sequence number, pan id, lladdr etc to a gnrc_netif_ieee802154 internal variables, or a proper layer, whichever is most feasible

Reply:
@gebart I would like to stick with the proposed solution of doing a `get()` call for the struct for now. I agree with your solution, but I'd rather not rework half of this PR at the moment, but do it in a follow up afterwards.

Reply:
Sounds good. A future possible TSCH implementation needs to use a different frame format than the traditional 802.15.4 frames use, so it would be a good improvement to decouple the framing from the device driver

Reply:
See https://github.com/RIOT-OS/RIOT/pull/9335#issuecomment-396889406. I think something is wrong with the concept here.... The layered approach *should* make it possible that it doesn't matter which pointer you use. And if it's not the case the pointer change should certainly not be implemented here IMHO, but in the netdev_ieee802154 adaption PR.